### PR TITLE
OpcodeDispatcher: Optimize addsubp{s,d} using fcadd 

### PR DESCRIPTION
**Pull request by Sonicadvance1**
This extension was added with seemingly Cortex-A710 and turns this
instruction in to two instructions which is quite good.

~~Needs https://github.com/FEX-Emu/FEX/pull/2994 merged first.~~

 Huge thanks to @dougallj for the optimization idea!

### CODE REVIEW

File: FEXCore/Source/Interface/Core/Interpreter/VectorOps.cpp [hunk 1]
```diff
@@ -2561,6 +2561,44 @@ DEF_OP(VPCMPISTRX) {
...
+  const uint8_t Elements = OpSize / ElementSize;
+
+  const auto Func = [Rotate](auto dst, auto src1, auto src2) {
+    auto Element1 = src2[1];
+    auto Element3 = src2[0];
+    if (Rotate == 90) {
+      Element1 = -Element1;
+    }
+    else {
+      Element3 = -Element3;
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/Interpreter/VectorOps.cpp:2585]: Element3 doesn't seem like it's used beyond this

File: FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp [hunk 2]
```diff
@@ -3825,6 +3825,50 @@ DEF_OP(VRev64) {
...
+  if (HostSupportsSVE256 && Is256Bit) {
+    const auto Mask = PRED_TMP_32B.Merging();
+
+    if (Dst == Vector1) {
+      // Trivial case where we already have source data to be averaged in
+      // the destination register. We can just do the operation in place.
+      fcadd(SubRegSize, Dst.Z(), Mask, Vector1.Z(), Vector2.Z(), Rotate);
+    }
+    else {
+      // SVE URHADD is a destructive operation, so we need
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp:3856]:
```suggestion
      // SVE FCADD is a destructive operation, so we need
```

File: FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp [hunk 3]
```diff
@@ -3825,6 +3825,50 @@ DEF_OP(VRev64) {
...
+    ElementSize == 2 ? ARMEmitter::SubRegSize::i16Bit :
+    ElementSize == 4 ? ARMEmitter::SubRegSize::i32Bit : ARMEmitter::SubRegSize::i64Bit;
+  const auto Rotate =
+    Op->Rotate == 90 ? ARMEmitter::Rotation::ROTATE_90 : ARMEmitter::Rotation::ROTATE_270;
+
+  if (HostSupportsSVE256 && Is256Bit) {
+    const auto Mask = PRED_TMP_32B.Merging();
+
+    if (Dst == Vector1) {
+      // Trivial case where we already have source data to be averaged in
```
Comment (lioncash) [inline FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp:3851]: The data doesn't seem to be averaged in this case

Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp]: Oop, updated comment.

File: FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp (hunk 2 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp]: Updated.

File: FEXCore/Source/Interface/Core/Interpreter/VectorOps.cpp (hunk 1 above)
Comment (Sonicadvance1) [reply FEXCore/Source/Interface/Core/Interpreter/VectorOps.cpp:2585]: Oops typo, fixed. Which is why CI failed.

**Review by lioncash — APPROVED** [review]

**Pull request merged**